kuka_iiqka_eac_driver/src/hardware_interface.cpp

```diff
@@ -42,7 +42,70 @@ CallbackReturn KukaEACHardwareInterface::on_init(const hardware_interface::Hardw
   hw_torque_commands_.resize(info_.joints.size(), 0.0);
   hw_stiffness_commands_.resize(info_.joints.size(), 30);
   hw_damping_commands_.resize(info_.joints.size(), 0.7);
+  hw_twist_commands_vector_.resize(6, 0.0);
+   
+if (info_.name=="KUKA_MR" ){
+  for (const hardware_interface::ComponentInfo & joint : info_.joints)
+  {
+    if (joint.command_interfaces.size() != 3)
+        {
+          RCLCPP_FATAL(
+            rclcpp::get_logger("KukaEACHardwareInterface"), "expecting exactly 3 command interface");
+          return CallbackReturn::ERROR;
+        }
+
+    if (joint.command_interfaces[0].name != hardware_interface::HW_IF_X)
+        {
+          RCLCPP_FATAL(
+            rclcpp::get_logger("KukaEACHardwareInterface"),
+            "expecting 'X' command interface as first");
+          return CallbackReturn::ERROR;
+        }
+    if (joint.command_interfaces[1].name != hardware_interface::HW_IF_Y)
+        {
+          RCLCPP_FATAL(
+            rclcpp::get_logger("KukaEACHardwareInterface"),
+            "expecting 'Y' command interface as second");
+          return CallbackReturn::ERROR;
+        }
+    if (joint.command_interfaces[2].name != hardware_interface::HW_IF_Z)
+        {
+          RCLCPP_FATAL(
+            rclcpp::get_logger("KukaEACHardwareInterface"),
+            "expecting 'Z' command interface as third");
+          return CallbackReturn::ERROR;
+        }
+    if (joint.state_interfaces.size() != 3)
+    {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("KukaEACHardwareInterface"), "expecting exactly 3 state interface");
+      return CallbackReturn::ERROR;
+    }
 
+    if (joint.state_interfaces[0].name != hardware_interface::HW_IF_X)
+    {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("KukaEACHardwareInterface"),
+        "expecting 'X' state interface as first");
+      return CallbackReturn::ERROR;
+    }
+    if (joint.state_interfaces[1].name != hardware_interface::HW_IF_Y)
+    {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("KukaEACHardwareInterface"),
+        "expecting 'Y' state interface as second");
+      return CallbackReturn::ERROR;
+    }
+    if (joint.state_interfaces[2].name != hardware_interface::HW_IF_Z)
+    {
+      RCLCPP_FATAL(
+        rclcpp::get_logger("KukaEACHardwareInterface"),
+        "expecting 'Z' state interface as third");
+      return CallbackReturn::ERROR;
+    }
+  }
+}
```

Comment:
these interfaces are not connected to an actual joint, so these check are not necessary (see my [comment ](https://github.com/kroshu/kuka_robot_descriptions/pull/56#discussion_r1588798220)in the robot_descriptions PR)